Core/Core/Router/Router.swift

```diff
@@ -152,7 +152,7 @@ public class Router: RouterProtocol {
     private func cleanURL(_ url: URLComponents) -> URLComponents {
         // URLComponents does all the encoding we care about except we often have + meaning space in query
         var url = url
-        url.query = url.query?.replacingOccurrences(of: "+", with: " ")
+        url.percentEncodedQuery = url.percentEncodedQuery?.replacingOccurrences(of: "+", with: "%20")
```

Comment:
This was causing some decoding of `%26` to `&` within a query param, which then caused it to move up to be a new query param. Using the `percentEncodedQuery` fixes this.